### PR TITLE
updated system tests for node.js dsm support

### PR DESCRIPTION
**Pull request by ida613**
## Description

New PR for updating system tests for node.js data stream monitoring support 

## Reviewer checklist

* [ ] If this PR modifies anything else than strictly the default scenario, then remove the `run-default-scenario` label
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Remove the `run-default-scenario` label to run all scenarios ([more info](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions))

Once your PR is reviewed, you can merge it! :heart:

### CODE REVIEW

**Comment by ida613** [comment]
We are getting these errors:
`denied: installation not allowed to Write organization package`
`ERROR: failed to solve: failed to push ghcr.io/datadog/system-tests/sample-app:6d08be7d12c306eb3ad954065fddb3f07667bf36: unexpected status: 403 Forbidden`
Is this a permission issue?

File: utils/build/docker/nodejs/express4/package.json
Comment (simon-id) [inline utils/build/docker/nodejs/express4/package.json:1]: you didn't use the npm install command so the package-lock.json file was not updated

**Review by simon-id — CHANGES_REQUESTED** [review]

File: tests/integrations/test_dsm.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This product includes software developed at Datadog (https://www.datadoghq.com/).
 # Copyright 2023 Datadog, Inc.
 
-from utils import weblog, interfaces, scenarios, released
+from utils import weblog, interfaces, scenarios, released, context
 from utils.tools import logger
```
Comment (cbeauchesne) [inline tests/integrations/test_dsm.py:8]: So far, the test is not activated. You must replace the `?` declared here for nodejs.

File: tests/integrations/test_dsm.py [hunk 2]
```diff
@@ -19,14 +19,16 @@ def setup_dsm_kafka(self):
     def test_dsm_kafka(self):
         assert self.r.text == "ok"
 
+        consumer_hash = 5700134256996109392 if context.library == "nodejs" else 4463699290244539355
```
Comment (cbeauchesne) [inline tests/integrations/test_dsm.py:22]: If each tracer has its own constant, this code will be easier to extend : 

```python
        if context.library == "nodejs":
            consumer_hash = 5700134256996109392 
            producer_hash = 13592025863925263556
        else:
            consumer_hash =  4463699290244539355
            producer_hash = 16645700936287432977
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/integrations/test_dsm.py (hunk 1 above)
Comment (ida613) [reply tests/integrations/test_dsm.py:8]: we were thinking we could add it after we do a release on dsm

Comment (cbeauchesne) [reply tests/integrations/test_dsm.py:8]: As you wish.

**Pull request merged**